commons/src/main/java/fr/free/nrw/commons/upload/ImageProcessing.java

```diff
@@ -0,0 +1,127 @@
+        String sel = MediaStore.Images.Media._ID + "=?";
+        Cursor cursor = context.getContentResolver().query(MediaStore.Images.Media.EXTERNAL_CONTENT_URI,
+                column, sel, new String[]{id}, null);
+
+        int columnIndex = cursor.getColumnIndex(column[0]);
+
+        if (cursor.moveToFirst()) {
+            filePath = cursor.getString(columnIndex);
+        }
+        cursor.close();
+
+        Log.d("Image", "File path: " + filePath);
+        return filePath;
+    }
+
+    //Extract GPS coords of image
+    public String getCoords(String filePath) {
+
+        ExifInterface exif;
+        String latitude = "";
+        String longitude = "";
+        String latitude_ref = "";
+        String longitude_ref = "";
+        String decimalCoords = "";
+
+        try {
+            exif = new ExifInterface(filePath);
+            latitude = exif.getAttribute(ExifInterface.TAG_GPS_LATITUDE);
+            latitude_ref = exif.getAttribute(ExifInterface.TAG_GPS_LATITUDE_REF);
+            longitude = exif.getAttribute(ExifInterface.TAG_GPS_LONGITUDE);
+            longitude_ref = exif.getAttribute(ExifInterface.TAG_GPS_LONGITUDE_REF);
+
+            Log.d("Image", "Latitude: " + latitude + " " + latitude_ref);
+            Log.d("Image", "Longitude: " + longitude + " " + longitude_ref);
+
+            decimalCoords = getDecimalCoords(latitude, latitude_ref, longitude, longitude_ref);
+
+        } catch (IOException e) {
+            Log.w("Image", e);
+        }
+        return decimalCoords;
+    }
+
+    //Converts format of coords into decimal coords as required by API for next step
+    private String getDecimalCoords(String latitude, String latitude_ref, String longitude, String longitude_ref) {
+
+        Float decLatitude, decLongitude;
+
+        if(latitude_ref.equals("N")){
+            decLatitude = convertToDegree(latitude);
+        }
+        else{
+            decLatitude = 0 - convertToDegree(latitude);
+        }
+
+        if(longitude_ref.equals("E")){
+            decLongitude = convertToDegree(longitude);
+        }
+        else{
+            decLongitude = 0 - convertToDegree(longitude);
+        }
+
+        return (String.valueOf(decLatitude) + ", " + String.valueOf(decLongitude));
+    }
+
+    private Float convertToDegree(String stringDMS){
+        Float result;
+        String[] DMS = stringDMS.split(",", 3);
+
+        String[] stringD = DMS[0].split("/", 2);
+        Double D0 = new Double(stringD[0]);
+        Double D1 = new Double(stringD[1]);
+        Double FloatD = D0/D1;
+
+        String[] stringM = DMS[1].split("/", 2);
+        Double M0 = new Double(stringM[0]);
+        Double M1 = new Double(stringM[1]);
+        Double FloatM = M0/M1;
+
+        String[] stringS = DMS[2].split("/", 2);
+        Double S0 = new Double(stringS[0]);
+        Double S1 = new Double(stringS[1]);
+        Double FloatS = S0/S1;
+
+        result = new Float(FloatD + (FloatM/60) + (FloatS/3600));
```

Comment:
Yeah, I think so. I have changed the Floats and Doubles to double where possible (barring the constructors that appear to be needed to convert the string values to doubles). I've made a new commit for that.